Add more documentation on the release process

pietroalbini opened the pull request:
This PR adds more documentation on the Ferrocene release process:

* It adds a "release process overview" page in the IP, detailing where the release process source code lives, how to publish a manual release, and how channels name are determined.
* It adds a "stable release process" page in the IP, outlining the manual steps required to publish a new stable release.
* It migrates the "releasing during outages" page from the legacy development guide to the IP (this should be the last page we needed to migrate).
* It updates the release information in the QP around channels and automated releases.

tshepang commented:
*Release manager(s)* is mentioned in a few places, but the role is not described anywhere

tshepang commented:
ferrocene/ci/channel in distributed tarballs does not match the official channel names as advertised... at the moment, they are all named `rolling` (in nightly, rolling, and pre-rolling tarballs)

pietroalbini commented:
> Release manager(s) is mentioned in a few places, but the role is not described anywhere

Documented this, and added cross-document links.

> ferrocene/ci/channel in distributed tarballs does not match the official channel names as advertised... at the moment, they are all named rolling (in nightly, rolling, and pre-rolling tarballs)

Yeah that's intended. I updated the wording on the channel names section, is that more clear?

tshepang commented on ferrocene/doc/internal-procedures/src/release/index.rst:
```diff
@@ -0,0 +1,100 @@
+the maintenance efforts.
+
+This way, as long as we set ``ferrocene/ci/channel`` to "rolling" on the main
+branch, we don't need make any change ourselves to promote a branch from
```
```suggestion
branch, we don't need to make any change ourselves to promote a branch from
```

tshepang reviewed (approved):
bors merge
